provider/terraform/resources/resource_storage_object_acl.go

```diff
@@ -56,198 +103,248 @@ func resourceStorageObjectAclCreate(d *schema.ResourceData, meta interface{}) er
 	bucket := d.Get("bucket").(string)
 	object := d.Get("object").(string)
 
-	predefined_acl := ""
-	role_entity := make([]interface{}, 0)
-
-	if v, ok := d.GetOk("predefined_acl"); ok {
-		predefined_acl = v.(string)
-	}
-
-	if v, ok := d.GetOk("role_entity"); ok {
-		role_entity = v.([]interface{})
-	}
+	// If we're using a predefined acl we just use the canned api.
+	if predefinedAcl, ok := d.GetOk("predefined_acl"); ok {
+		res, err := config.clientStorage.Objects.Get(bucket, object).Do()
```

Comment:
This read is unnecessary, right?

Reply:
Nope! We use it a few lines down as part of `clientStorage.Objects.Update(bucket, object, res)`.

Reply:
oh yup there it is, thanks!